### PR TITLE
Update IP allowlist formation, allow adding IP addresses/networks via predefined groups

### PR DESCRIPTION
**Pull request by mattops**

### CODE REVIEW

File: charts/generic-service/values.yaml [hunk 1]
```diff
@@ -82,6 +82,33 @@ ingress:
        t:none,\
        setvar:tx.paranoia_level=2"
 
+# IP allowlist to be applied via ingress-nginx.
+allowlist: {}
+# Version of allowlist being applied (supplied by CI/CD)
+allowlist_version: ""
```
Comment (andrewrlee) [inline charts/generic-service/values.yaml:88]: is this just for auditing purposes?

Comment (mattops) [reply charts/generic-service/values.yaml:88]: Yes, so we can track what version of the allowlist has been applied.

File: charts/generic-service/ci/test-values.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (andrewrlee) [inline charts/generic-service/ci/test-values.yaml:1]: is this file needed?

File: charts/generic-service/templates/_helpers.tpl [hunk 3]
```diff
@@ -55,9 +55,45 @@ release: {{ .Release.Name }}
...
+    {{- if and (ne $k "groups") (kindIs "string" $v) -}}
+      {{- $allAllowlists = append $allAllowlists $v -}}
+      {{- $allCustomItemNames = append $allCustomItemNames $k -}}
+    {{- end -}}
+    {{- if eq $k "groups" -}}
+      {{- $_ := required "'groups' key found in 'allowlist' values, so 'allowlist_groups' is a required value but was not supplied." $.Values.allowlist_groups }}
+      {{- $groups := $v -}}
+      {{- range $group := $groups -}}
+        {{- if not (get $.Values.allowlist_groups $group) -}}
+          {{- required "Invalid group found in 'allowlist', check defined lists in 'allowlist_groups'." "" }}
```
Comment (andrewrlee) [inline charts/generic-service/templates/_helpers.tpl:75]: not required now but as a polishing step, could potentially include list of available groups in log message

**Review by andrewrlee — APPROVED** [review]
LGTM

File: charts/generic-service/templates/_helpers.tpl (hunk 3 above)
Comment (mattops) [reply charts/generic-service/templates/_helpers.tpl:75]: yes, thought might add that once we know for sure what they are.
